app/views/admin_sets/_roles_tables.html.erb

```diff
@@ -0,0 +1,33 @@
+<table class='table table-striped'>
+  <thead class='thead-dark'>
+    <tr>
+      <th scope='col'> Viewers </th>
+    </tr>
+  </thead>
+  <tbody>
+    <% User.all.each do |user| %>
```

Comment:
Instead of iterating over all of the users you can do this:
User.with_role(:viewer, @admin_set)
It runs a single join query to get all of the users who have the specified role.

Looking at it on screen, it would also be useful to sort the list (though this wasn't in the issue):
User.with_role(:viewer, @admin_set).order('last_name ASC')